### PR TITLE
Unit Test Cases For the the Viewmodel

### PR DESCRIPTION
**Pull request by Vanshbajaj**
Unit Test Cases  added for the the Viewmodel

### CODE REVIEW

File: presentation/src/test/java/com/practical/presentation/CharacterViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+
+@ExperimentalCoroutinesApi
+@OptIn(ExperimentalCoroutinesApi::class)
+class CharacterViewModelTest {
+    @get:Rule
+    val instantExecutorRule = InstantTaskExecutorRule()
```
Comment (shreyansh-mm) [inline presentation/src/test/java/com/practical/presentation/CharacterViewModelTest.kt:34]: Not needed

File: presentation/src/test/java/com/practical/presentation/CharacterViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    private lateinit var viewModel: CharacterViewModel
+    private lateinit var getCharactersUseCase: GetCharactersUseCase
+    private val testDispatcher = StandardTestDispatcher()
+
+    @Before
+    fun setUp() {
+        getCharactersUseCase = mockk(relaxed = true)
+        Dispatchers.setMain(testDispatcher)
```
Comment (shreyansh-mm) [inline presentation/src/test/java/com/practical/presentation/CharacterViewModelTest.kt:42]: I don't think its needed

File: presentation/src/test/java/com/practical/presentation/CharacterViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        viewModel = CharacterViewModel(getCharactersUseCase, testDispatcher)
+    }
+
+    @After
+    fun tearDown() {
+        Dispatchers.resetMain()
```
Comment (shreyansh-mm) [inline presentation/src/test/java/com/practical/presentation/CharacterViewModelTest.kt:48]: I don't think this is needed either

File: presentation/src/test/java/com/practical/presentation/CharacterViewModelTest.kt [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    @Test
+    fun `given the ViewModel is initialized, when fetching characters, then it should emit loading state initially`() {
+        // Given
+        // ViewModel is initialized
+        viewModel.fetchCharacters()
```
Comment (shreyansh-mm) [inline presentation/src/test/java/com/practical/presentation/CharacterViewModelTest.kt:55]: View model is already initialized, we don't need this

**Review by shreyansh-mm — CHANGES_REQUESTED** [review]

**Review by shubhamsinghmutualmobile — APPROVED** [review]

**Pull request merged**